### PR TITLE
[search] Added hotel rating and hotel pricing to result metadata.

### PR DESCRIPTION
**Pull request by VladiMihaylenko**
@ygorshenin @mgsergio ptal

### CODE REVIEW

File: search/intermediate_result.cpp [hunk 1]
```diff
@@ -52,9 +53,26 @@ void ProcessMetadata(FeatureType const & ft, Result::Metadata & meta)
...
+  if (isSponsoredHotel)
+  {
+    auto const r = src.Get(feature::Metadata::FMD_RATING);
+    char const * rating = r.empty() ? kEmptyRatingSymbol : r.c_str();
+    meta.m_hotelRating = rating;
+
+    int pricing;
+    strings::to_int(src.Get(feature::Metadata::FMD_PRICE_RATE), pricing);
+    string pricingStr;
+    CHECK(pricing >= 0, ("Pricing must be positive!"));
```
Comment (ygorshenin) [inline search/intermediate_result.cpp]: CHECK_GREATER_OR_EQUAL

File: search/intermediate_result.cpp [hunk 2]
```diff
@@ -52,9 +53,26 @@ void ProcessMetadata(FeatureType const & ft, Result::Metadata & meta)
...
-  meta.m_isSponsoredHotel = ftypes::IsBookingChecker::Instance()(ft);
-  
+
+  bool const isSponsoredHotel = ftypes::IsBookingChecker::Instance()(ft);
+  meta.m_isSponsoredHotel = isSponsoredHotel;
+
+  if (isSponsoredHotel)
+  {
+    auto const r = src.Get(feature::Metadata::FMD_RATING);
+    char const * rating = r.empty() ? kEmptyRatingSymbol : r.c_str();
```
Comment (ygorshenin) [inline search/intermediate_result.cpp]:
```c++
char const * const ...
```

**Comment by mgsergio** [comment]
LGTM

**Comment by ygorshenin** [comment]
LGTM after fixes.

**Pull request merged**